.github/workflows/shared-build-and-test.yaml

```diff
@@ -11,6 +11,14 @@ jobs:
       - name: Checkout repo
         uses: actions/checkout@v3
 
+      - name: Set up JDK
```

Comment:
How does this work? Which version will the mvn step use?

Reply:
We specify the JDK version in the relevant project's pom.xml file under `maven-compiler-plugin`. For example:
```
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.8.0</version>
                <configuration>
                    <source>11</source>
                    <target>11</target>
                </configuration>
            </plugin>
```